x/tx/signing/get_signers.go

```diff
@@ -50,6 +54,43 @@ func getSignersFieldNames(descriptor protoreflect.MessageDescriptor) ([]string,
 	return signersFields, nil
 }
 
+// init performs a dry run of getting all msg's signers. This has 2 benefits:
+// - it will error if any Msg has forgotten the "cosmos.msg.v1.signer"
+// annotation
+// - it will pre-populate the context's internal cache for getSignersFuncs
+// so that calling it in antehandlers will be faster.
+func (c *GetSignersContext) init() error {
+	fs, ok := c.protoFiles.(*protoregistry.Files)
+	if !ok {
+		return fmt.Errorf("expected *protoregistry.Files in GetSignersContext, got %T", c.protoFiles)
+	}
```

Comment:
I had to add this change after #15413.

@aaronc @kocubinski In practice, would we ever want `protodesc.Resolver` not to be `*protoregistry.Files`?

Reply:
Maybe for testing failure modes in unit tests?